core/src/main/scala/cats/Foldable.scala

```diff
@@ -56,6 +57,8 @@ import simulacrum.typeclass
       }
     }
 
+  def size[A](fa: F[A]): Long = foldMap(fa)(_ => 1)
```

Comment:
👍 I think that `Long` was a good choice here. `Int` is probably the first thing that would have come to my mind, but people could conceivably be using `Foldable` for some very large structures. Though hopefully if their structure is large enough to pass the max `Int` value they have overridden `size` to a more efficient implementation :)